### ISSUE TITLE
Version Mismatch

### ISSUE THREAD

**Issue by Findarato** [lajg-dev/Obsidian-Plugin-GPG-Inline-Encrypt#5]
When I try and use the tool I get a warning saying my version is not the same as the one the tool is using.

```bash
gpg --version
gpg (GnuPG) 2.4.5
libgcrypt 1.11.0-unknown
Copyright (C) 2024 g10 Code GmbH
License GNU GPL-3.0-or-later <https://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Home: /home/joe/.gnupg
Supported algorithms:
Pubkey: RSA, ELG, DSA, ECDH, ECDSA, EDDSA
Cipher: IDEA, 3DES, CAST5, BLOWFISH, AES, AES192, AES256, TWOFISH,
        CAMELLIA128, CAMELLIA192, CAMELLIA256
Hash: SHA1, RIPEMD160, SHA256, SHA384, SHA512, SHA224
Compression: Uncompressed, ZIP, ZLIB, BZIP2
```
The tool is asking for 2.5.1 but that is still the development release on the GNUGP website.
https://www.gnupg.org/download/

The mailing list has that version set for public testing, and not general availability. 

https://lists.gnupg.org/pipermail/gnupg-announce/2024q3/000485.html

Could the version be lowered back to the current stable version please.

**Comment by Findarato** [comment]
Just to add on, this only happens when trying to do inline GPG, the whole document GPG works fine

**Comment by kbenton** [comment]
I have an even weirder and worse situation. Inline encrypt fails, and full document encrypt succeeds, however then it fails to *decrypt* the encrypted full document (with essentially the same error). It's just test text so no big deal and I could recover it by working on the command line, but needless to say I can't quite trust the plugin at the moment. 

The error I get complains specifically about keyboxd on Linux Mint. It's hard to capture the full error since the popup disappears, but it's basically 'server keyboxd is older than us (2.4.4 vs. 2.5.1)'. I'm actually not sure what the "us" is that's at version 2.5.1 in this context... the installed version of gpg at the path specified in the plugin config is 2.4.4. Nothing in the plugin code appears to check or care what version of gpg is being used, only if it exists at the path specified.

**Comment by lajg-dev** [comment]
First of all, I want to thank you for using the plugin and for reporting this type of inconvenience. I would like to know a little more about the error that is occurring. If you could describe what OS you are using and if you could include some kind of screenshot or video of the error, it would be very useful to replicate it in a controlled environment and solve this problem.
I am sorry that this is happening with the plugin.
As soon as I can reproduce the error, I will work to fix it and leave a stable version.

**Comment by kbenton** [comment]
Yep, totally. And to be fair the error does seem to be coming thru from gpg, not the plugin itself. I'm on Linux Mint 22 "Wilma", all updates current. I've attached screenshots of both the inline encrypt error and the decrypt error, which has more components that are being referenced (including pinentry & gpg-agent).

I've tested gpg in the terminal and all seems fine, including the pinentry dialog, and i've still yet to find anything that corresponds to wanting to be version 2.5.1. (Note that I have tried the suggested gpgconf --kill all command among other troubleshooting.)
![decrypt](https://github.com/user-attachments/assets/33013261-343c-488b-a4e4-bf384237800f)
![inline enc](https://github.com/user-attachments/assets/c1fed00a-12cc-4615-b72b-c8555846c64d)

**Comment by lajg-dev** [comment]
Thank you very much for taking the time to document more about this error, could we validate which gpg command is being executed at the time of the error?

You can find out this by enabling the developer console in the following option

![Screenshot 2024-12-04 at 10 21 25 AM](https://github.com/user-attachments/assets/c9a15e58-fbdb-4d05-9b33-e554bbd322fb)

Then you enable the configuration to print commands to the console in the plugin:

![Screenshot 2024-12-04 at 10 21 42 AM](https://github.com/user-attachments/assets/d2ac3b05-4efb-4275-9e8f-987ec6bafde0)

And finally, if you can test that same command outside the obsidian window, then we know the command and rule out that the problem is with the obsidian component, once the error is identified, we can look for a way to solve it.

**Comment by Findarato** [comment]
I do not have the option to turn on the Developer console outputs. When I try the inline I still get the same and the dev console shows nothing
![image](https://github.com/user-attachments/assets/37ff427d-4a86-4502-a203-0167a943a7ae)

**Comment by lajg-dev** [comment]
Please review last update [1.1.5](https://github.com/lajg-dev/Obsidian-Plugin-GPG-Inline-Encrypt/releases/tag/1.1.5), here I fix some bugs, including the reported in this issue (Additionally, enable the plugin for the edit mode), please check and let me know if this issue could be closed.

**Comment by lajg-dev** [comment]
I am closing this ticket because I haven't received a response in more than 3 months since the change was made. I assume the change fixed the original issue. If you have any further questions, please feel free to contact me.
